presets/ragengine/main.py

```diff
@@ -72,15 +142,68 @@ async def query_index(request: QueryRequest):
             status_code=500, detail=f"An unexpected error occurred: {str(e)}"
         )
 
-@app.get("/indexes", response_model=List[str])
+@app.get(
+    "/indexes",
+    response_model=List[str],
+    summary="List All Indexes",
+    description="""
+    Retrieve the names of all available indexes.
+
+    ## Response Example:
+    ```json
+    [
+      "example_index",
+      "test_index"
+    ]
+    ```
+    """,
+)
 def list_indexes():
     try:
         return rag_ops.list_indexes()
     except Exception as e:
         raise HTTPException(status_code=500, detail=str(e))
 
-@app.get("/indexes/{index_name}/documents", response_model=ListDocumentsResponse)
-async def list_documents_in_index(index_name: str):
+@app.get(
+    "/indexes/{index_name}/documents",
+    response_model=ListDocumentsResponse,
+    summary="List Documents in an Index",
+    description="""
+    Retrieve a paginated list of documents for a given index.
+
+    ## Request Example:
+    ```
+    GET /indexes/test_index/documents?limit=5&offset=5&max_text_length=500
+    ```
+
+    ## Response Example:
+    ```json
+    {
+      "documents": [
+        {
+          "doc_id": "123456",
+          "text": "Sample document text.",
+          "metadata": {"author": "John Doe"},
+          "is_truncated": true
+        },
+        {
+          "doc_id": "123457",
+          "text": "Another document text.",
+          "metadata": {"author": "Jane Doe"},
+          "is_truncated": false
+        }
+      ],
+      "count": 5
+    }
+    ```
+    """,
+)
+async def list_documents_in_index(
+    index_name: str,
+    limit: int = Query(10, ge=1, le=100, description="Maximum number of documents to return"),
+    offset: int = Query(0, ge=0, description="Starting point for the document list"),
+    max_text_length: Optional[int] = Query(1000, ge=1, description="Maximum text length to return"),
```

Comment:
max_text_length means the max text length returned per document. We need to make it clear.  Otherwise, user may think it is the max_length for all documents

Reply:
ok I fixed the description, I think thats sufficient but if you think name change is needed happy to do that

Reply:
could rename it to `max_text_length_per_document` just find it wordy if we can avoid